R/sources_to_targets.R

```diff
@@ -11,43 +11,54 @@
 #'
 #' @examples
 #' if (FALSE) {
-#' andorra_la_vella = c(1.5218, 42.5075)
-#' pas_de_la_casa = c(1.7333, 42.5425)
-#' # Another popular location in andorra:
-#' encamp = c(1.5763, 42.5343)
-#' from = rbind(andorra_la_vella, encamp)
-#' to = rbind(pas_de_la_casa)
-#' sources_to_targets(from, to)
+#'   andorra_la_vella <- c(1.5218, 42.5075)
+#'   pas_de_la_casa <- c(1.7333, 42.5425)
+#'   # Another popular location in andorra:
+#'   encamp <- c(1.5763, 42.5343)
+#'   from <- rbind(andorra_la_vella, encamp)
+#'   to <- rbind(pas_de_la_casa)
+#'   sources_to_targets(from, to)
 #' }
-sources_to_targets = function(
+sources_to_targets <- function(
     from,
     to,
     costing = "pedestrian",
     directions_options = list(units = "km"),
     url = "http://localhost:8002",
-    ...
-) {
-  
-  n_from = nrow(from)
-  n_to = nrow(to)
-  params = list(
+    ...) {
+  n_from <- nrow(from)
+  n_to <- nrow(to)
+  params <- list(
     sources = lapply(seq_len(n_from), create_coordinate_list, data = from),
     targets = lapply(seq_len(n_to), create_coordinate_list, data = to),
     costing = costing,
     directions_options = directions_options,
     ...
   )
-  json = vh_get(resource="sources_to_targets",
-                from, 
-                to,
-                costing, 
-                units,
-                url = url, 
-                params = params)
-  
-  do.call(rbind, lapply(json$sources_to_targets, function(x) as.data.frame(x[[1]])))
+  json <- vh_get(
+    resource = "sources_to_targets",
+    from,
+    to,
+    costing,
+    units,
+    url = url,
+    params = params
+  )
+
+  do.call(rbind, lapply(json$sources_to_targets[[1]], function(x) as.data.frame(x))) |>
```

Comment:
OK so this adds new columns with from and to lat and lon right? What's the use case, why can't the user pull out with `st_startpoint()` or similar?

Just want to understand the intention here..